python/example_code/auto-scaling/action_wrapper.py

```diff
@@ -53,91 +61,150 @@ def create_group(
+                    "Consider deleting unused resources or request a limit increase. "
+                    "\nSee Auto Scaling Service Quota documentation here:"
+                    "\n\thttps://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-quotas.html"
+                )
+            logger.error(f"Full error:\n\t{err}")
             raise
 
     # snippet-end:[python.example_code.auto-scaling.CreateAutoScalingGroup]
 
     # snippet-start:[python.example_code.auto-scaling.UpdateAutoScalingGroup]
-    def update_group(self, group_name, **kwargs):
+    def update_group(self, group_name: str, **kwargs: Any) -> None:
         """
         Updates an Auto Scaling group.
 
         :param group_name: The name of the group to update.
         :param kwargs: Keyword arguments to pass through to the service.
+        :return: None
+        :raises ClientError: If there is an error updating the Auto Scaling group.
         """
         try:
             self.autoscaling_client.update_auto_scaling_group(
                 AutoScalingGroupName=group_name, **kwargs
             )
+            logger.info(f"Successfully updated Auto Scaling group {group_name}.")
+
         except ClientError as err:
-            logger.error(
-                "Couldn't update group %s. Here's why: %s: %s",
-                group_name,
-                err.response["Error"]["Code"],
-                err.response["Error"]["Message"],
-            )
+            error_code = err.response["Error"]["Code"]
+            logger.error(f"Failed to update Auto Scaling group {group_name}.")
+            if error_code == "ResourceInUse":
+                logger.error(
+                    "The Auto Scaling group '%s' is currently in use and cannot be modified. Please try again later.",
+                    group_name,
+                )
+            elif error_code == "ScalingActivityInProgress":
+                logger.error(
+                    f"A scaling activity is currently in progress for the Auto Scaling group '{group_name}'."
+                    "Please wait for the activity to complete before attempting to update the group."
+                )
+            logger.error(f"Full error:\n\t{err}")
             raise
 
     # snippet-end:[python.example_code.auto-scaling.UpdateAutoScalingGroup]
 
     # snippet-start:[python.example_code.auto-scaling.DeleteAutoScalingGroup]
-    def delete_group(self, group_name):
+    def delete_group(self, group_name: str) -> None:
         """
         Deletes an Auto Scaling group. All instances must be stopped before the
         group can be deleted.
 
         :param group_name: The name of the group to delete.
+        :return: None
+        :raises ClientError: If there is an error deleting the Auto Scaling group.
         """
         try:
             self.autoscaling_client.delete_auto_scaling_group(
                 AutoScalingGroupName=group_name
             )
+
+            # Wait for the group to be deleted
```

Comment:
Punctuation on comments (throughout).

Reply:
I just saw 2 instances of this in this file. Fixed both.